arroyo-sql/src/expressions.rs

```diff
@@ -2151,3 +2185,56 @@ impl JsonExpression {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct RustUdfExpression {
+    name: String,
+    args: Vec<(TypeDef, Expression)>,
+    ret_type: TypeDef,
+}
+
+impl RustUdfExpression {
+    fn to_syn_expression(&self) -> syn::Expr {
+        let name = format_ident!("{}", &self.name);
+
+        let (defs, args): (Vec<_>, Vec<_>) = self
+            .args
+            .iter()
+            .enumerate()
+            .map(|(i, (def, expr))| {
+                let t = expr.to_syn_expression();
+                let id = format_ident!("__{}", i);
+                let def = match (def.is_optional(), expr.nullable()) {
+                    (true, true) | (false, false) => quote!(let #id = #t),
+                    (true, false) => quote!(let #id = Some(#t)),
+                    (false, true) => quote!(let #id = (#t)?),
+                };
+                (def, quote!(#id))
+            })
+            .unzip();
+
+        let mut ret = quote!(udfs::#name(#(#args, )*));
+
+        if self.return_type().is_optional() && !self.ret_type.is_optional() {
+            // we have to wrap the result in Some
+            ret = quote! { Some(#ret) }
+        };
+
+        parse_quote!({
+            (|| {
+                #(#defs; )*
+                #ret
+            })()
+        })
+    }
+
+    fn return_type(&self) -> TypeDef {
+        self.ret_type.with_nullity(
+            self.ret_type.is_optional()
```

Comment:
This seems off, I think the return type should be null if either
* The UDF returns null
* There are non-null UDF arguments with nullable arguments.

If I have 
```
fn my_square(x: Option<i64>) -> i64 {
  x.map(|x| x * x).unwrap_or(0)
}
```
And I run 
```
select my_square(bid.price) FROM nexmark;
```
Then the result is correctly not nullable. However, if you run
```
select my_square(5) FROM nexmark;
```
The result is wrapped in Some().